### ISSUE TITLE
[Export] Improve creatibutor name export 

### ISSUE THREAD

**Issue by fenekku** [galterlibrary/digital-repository#1024]
**[Edits 2022-06-24 to take [comment #1](https://github.com/galterlibrary/digital-repository/issues/1024#issuecomment-1164822163) and [comment #2](https://github.com/galterlibrary/digital-repository/issues/1024#issuecomment-1164824541) into account]**
 
There are issues with name exports:

- [x] Creators/contributors with suffixes (e.g. Jr.) are not exported correctly (1)
- [x] ... with additional commas are not exported correctly (2)
- [x] ... with dates in their names  are not exported correctly (3)
- [ ] Some creators are not exported at all  (4)

Examples:

1) in DH: "Miller, Alfred Frederick Jr." and in Prism: "Miller, Alfred Frederick"
2) in DH: "Mckay, Frederick, S." and in Prism: "Mckay, Frederick".
3) in DH: "Mason, Michael L., 1895-1963" and in Prism: "Mason, Michael L."
4) in DH:  (https://digitalhub.northwestern.edu/files/50685d7b-0d3f-4b1f-9f97-a5d1a5b6d74e)
Daly, Elyse Renee
Battalio, Sam
Spring, Bonnie
Wafford, Q. Eileen
Pfammatter, Angela Fidler
Jovanovic, Christine Ernestine Schaak
Alshurafa, Nabil
Kalam, Faiza
Stump, Tammy Kay 

and in Prism

Daly, Elyse Renee
Battalio, Sam
Spring, Bonnie

The export for these kinds of cases should be:

1)
```json
{
    "person_or_org": {
        "family_name": "Miller",
        "given_name": "Alfred Frederick Jr.",
        "type": "personal"
    }
},
```
On UI: "Miller, Alfred Frederick Jr."

2)
```json
{
    "person_or_org": {
        "family_name": "Mckay",
        "given_name": "Frederick S.",
        "type": "personal"
    }
},
```
On UI: "Mckay, Frederick S."

3)

```json
{
    "person_or_org": {
        "family_name": "Mason",
        "given_name": "Michael L., 1895-1963",
        "type": "personal"
    }
},
```
On UI: "Mason, Michael L., 1895-1963"

4) should have all the creators

So that the full information is preserved.

**Comment by gneidhardt** [comment]
I actually would request to leave the comma between the name and date - that's how the names are written out in LCNAF format so I think it makes sense to match that. Otherwise looks good to me.

**Comment by gneidhardt** [comment]
Not sure if this is the right issue or should create another, but wanted to note this issue too: in some instances, uploader names are not in Prism (example: https://digitalhub.northwestern.edu/files/50685d7b-0d3f-4b1f-9f97-a5d1a5b6d74e, https://vtfsmghslrepo02.fsm.northwestern.edu/records/x7rh0-ys603)

**Comment by fenekku** [comment]
Edits made. Maybe @gneidhardt another look at comma placement?

**Comment by Meowcenary** [comment]
Example of file with "Miller, Alfred Frederick Jr." - https://digitalhub.northwestern.edu/files/432e29fb-ce35-487f-be6b-195ba9eb389f

**Comment by Meowcenary** [comment]
Until I figure out an addon for pointing tickets I'm just going to comment. I think this is a **3 point** ticket. I'm a bit blocked by how slow the rails console is to check why users are not being found, but once that's in place I think the fix should be easy enough.

**Comment by Meowcenary** [comment]
PR is up for lazy loading searchable files which will hopefully unblock me from testing more here. Doubt it will be released before end of day though.

**Comment by Meowcenary** [comment]
I think I finally figured out the issue with the missing creators / contributors. The formal names that were being used to lookup users expected a middle name for the user. In the example record the file does not include the middle name for all creators, but I was able to use the data to find them with their display name (i.e first name last name). 

Sort of related - I'm going to blow the cache for memoized creators_contributors with this commit too since I ran into some issues with merge conflicts.
